File: sentiment_server/apps/analysis/application/queries/runtime_capabilities.py
from django.conf import settings

from apps.admin_panel.domain.rules.dataset_import import (
    DATASET_IMPORT_SUPPORTED_FORMATS,
)
from apps.admin_panel.infra.automation.system_config import load_system_config
from apps.analysis.infra.file_parsing import BATCH_ANALYSIS_SUPPORTED_FORMATS


def build_runtime_capabilities_payload():
    config = load_system_config()
    return {
        "max_upload_size_mb": round(
            int(getattr(settings, "MAX_UPLOAD_SIZE", 0)) / (1024 * 1024), 2
        ),
        "batch_analysis_supported_formats": list(BATCH_ANALYSIS_SUPPORTED_FORMATS),
        "dataset_import_supported_formats": list(DATASET_IMPORT_SUPPORTED_FORMATS),
        "report_defaults": {
            "report_type": config.get("default_report_type", "weekly"),
            "report_format": config.get("default_report_format", "pdf"),
        },
    }

File: sentiment_webapp/src/utils/runtimeCapabilities.js
export const FALLBACK_RUNTIME_CAPABILITIES = {
  max_upload_size_mb: 10,
  batch_analysis_supported_formats: ['.xlsx', '.txt'],
  dataset_import_supported_formats: ['.xlsx', '.txt'],
  report_defaults: {
    report_type: 'weekly',
    report_format: 'pdf',
  },
}

/**
 * @param {Record<string, any>} [payload={}]
 */
export function normalizeRuntimeCapabilities(payload = {}) {
  const batchFormats =
    payload.batch_analysis_supported_formats ??
    FALLBACK_RUNTIME_CAPABILITIES.batch_analysis_supported_formats
  const datasetFormats =
    payload.dataset_import_supported_formats ??
    FALLBACK_RUNTIME_CAPABILITIES.dataset_import_supported_formats

  return {
    max_upload_size_mb:
      payload.max_upload_size_mb ?? FALLBACK_RUNTIME_CAPABILITIES.max_upload_size_mb,
    batch_analysis_supported_formats: [...batchFormats],
    dataset_import_supported_formats: [...datasetFormats],
    report_defaults: {
      report_type:
        payload.report_defaults?.report_type ??
        FALLBACK_RUNTIME_CAPABILITIES.report_defaults.report_type,
      report_format:
        payload.report_defaults?.report_format ??
        FALLBACK_RUNTIME_CAPABILITIES.report_defaults.report_format,
    },
  }
}
